src/test/java/core/basesyntax/service/FruitParserTest.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.model.Fruit;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitParserTest {
+    private static FruitParser parser;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        parser = new FruitParser();
+    }
+
+    @Test
+    public void parser_correctData_OK() {
+        String data = "s,apple,13";
+        Transaction expected = new Transaction(Transaction.Operation.SUPPLY,
+                new Fruit("apple"), 13);
+        Transaction actual = parser.parseData(data);
+        Assert.assertEquals(expected, actual);
+    }
```

Comment:
I think we can add here test for invalid data too

Reply:
We don't have to test Parser for invalid data, becouse first we use validator which is already tested for invalid data

Reply:
Sometimes it's useful to test how services work in conjunction. Perhaps your service can change data somehow before it passed to another one